### ISSUE TITLE
Set up resizing/cropping for `gr.Image` uploads

### ISSUE THREAD

**Issue by apolinario** [gradio-app/gradio#7929]
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
Sometimes models only accept a specific width/height dimension and/or proportion as inputs. Currently users have to build this logic on their app.

**Describe the solution you'd like**  
Add to the `gr.Image` component tooling resizing/cropping to user specified dimensions. As a high level first API suggestion, maybe there could be a width and height limit, and a param that either resizes or crops to that size

**Comment by pngwn** [comment]
This is supported by the image editor and was removed from the `Image` component intentionally. 

Perhaps we should make sure the `ImageEditor` component can support this usecase by making sure everything else can be disabled.

**Comment by apolinario** [comment]
To clarify my original suggestion: 

Many models take in an image that needs to be, specifically, say, 1280x720

If a user uploads an arbitrarily sized image, I'd break ofc. So here what I suggested is some setting like `target_size=1024` and then `target_crop=True` would crop the image to this proportion (no stretching), and `False` would instead resize the image to the proportion even if involves some stretching. 

Originally, I thought of this happening on the backend, no front-end editing (hence the `gr.Image` suggestion, not the editor).

I think the Editor could handle that too, with a UX that maybe lets visually the user choose where to crop/resize (similar to the Twitter UI), but I still don't hate the idea of dealing with it in the backend only

**Comment by abidlabs** [comment]
Yeah we could add back a `target_size` parameter to both `gr.Image` and `gr.ImageEditor`. We initially removed it from `gr.Image` (back when the parameter was called `shape`) because it was confusing for users to have `width` and `height` parameters (which controlled the size of the `gr.Image` component in the Gradio UI) as well as to have a `shape` parameter. But I like `target_size` as the name is more clear. Just a few clarification questions @apolinario:

* I'm surprised we haven't heard more user requests for this parameter, and I ascribed it to the fact that perhaps increasingly models are able to process images of different sizes. Is that true? Or do you still see a lot of vision models that only work with images of fixed sizes?
* The `target_size` parameter should be a tuple of `(width, height)` for the more general case?
* Perhaps the `gr.Image()` could just apply a resize-and-center-crop-based method of resizing an image when `target_size` is specified, whereas `gr.ImageEditor()` could provide a UI for a user to choose what part of the input image to keep (when the inputted-image's aspect ratio is not the same as the `target_size`, I think you had suggested this earlier @pngwn)

**Comment by pngwn** [comment]
The image editor already has canvas size, so I don't think we need two sets of w.h kwargs. That would just be confusing. We just need to present the crop window in response to uploading an image to give the user a choice on how to handle the disparity if the canvas is fixed. Maybe via a flag.

**Comment by apolinario** [comment]
> I'm surprised we haven't heard more user requests for this parameter, and I ascribed it to the fact that perhaps increasingly models are able to process images of different sizes. Is that true? Or do you still see a lot of vision models that only work with images of fixed sizes?

Indeed I think from when I created the issue to today a few more models accept more resolutions, however not rarely it is still a set of fixed resolutions (e.g.: either a square, landscape or portrait). But this would be too complex of a UX maybe

**Comment by pngwn** [comment]
We have a dedicated crop UI now, we could accept a series of valid sizes for image editor at least. 

For image it might be tricky and really starts to stretch the meaning of 'image' if we did that.

**Comment by abidlabs** [comment]
We're deprioritizing enhancements to components since users can create custom components or even completely custom frontends using `gr.Server`: https://huggingface.co/blog/introducing-gradio-server
